capstone/src/main/java/com/google/sps/data/dexparser/DexData.java

```diff
@@ -0,0 +1,173 @@
+		parseHeaderItem();
+		
+	}
+	
+	
+	/*
+	 * Checks that the magic number is valid
+	 */
+	public boolean verifyMagicNumber(byte[] magicNumber) {
+		return Arrays.equals(magicNumber, HeaderItem.DEX_FILE_MAGIC_v035)
+				|| Arrays.equals(magicNumber, HeaderItem.DEX_FILE_MAGIC_v037)
+				|| Arrays.equals(magicNumber, HeaderItem.DEX_FILE_MAGIC_v038)
+				|| Arrays.equals(magicNumber, HeaderItem.DEX_FILE_MAGIC_v039);
+	}
+
+	public void parseHeaderItem() throws IOException, DataFormatException {
+		mHeaderItem = new DexStructure.HeaderItem();
+
+		seek(0);
+
+		byte[] magicNumber = new byte[8];
+
+    try {
+      mDexFile.readFully(magicNumber);
+    } catch (EOFException e) {
+      System.out.println("");
+      LOGGER.info("End of file reached");
+    } catch (IOException e) {}
+
+		// Checks the magic number
+		if (!verifyMagicNumber(magicNumber)) {
+			LOGGER.info("Magic number is wrong. Are you sure this is a dex file?");
+			throw new DataFormatException();
+		} else {
+      LOGGER.info("Magic number has been verified.");
+		}
+
+		/*
+		 * Skip the magic number, checksum, signature, file size, header size and place
+		 * pointer at the beginning of endian tag to get endianess. Offset: 40
+		 */
+		seek(8 + 4 + 20 + 4 + 4);
+
+		mHeaderItem.setEndianTag(readInt());
+		LOGGER.info("Endian Tag: " + mHeaderItem.getEndianTag());
+
+		if (mHeaderItem.getEndianTag() == HeaderItem.ENDIAN_CONSTANT) {
+      LOGGER.info("DEX file is based on Little Endian.");
+		} else if (mHeaderItem.getEndianTag() == HeaderItem.REVERSE_ENDIAN_CONSTANT) {
+      LOGGER.info("DEX file is based on Big (Reverse) Endian.");
+		} else {
+      LOGGER.info("Endian Constant has unexpected value " + Integer.toHexString(mHeaderItem.getEndianTag()));
+			throw new DataFormatException();
+		}
+
+		// set offset after magic, checksum, signature
+		seek(8 + 4 + 20);
+		ByteBuffer buffer = readByteBuffer(Integer.BYTES * 20);
+		mHeaderItem.setFileSize(buffer.getInt());
+		mHeaderItem.setHeaderSize(buffer.getInt());
+		/*mHeaderItem.endianTag =*/ buffer.getInt(); // Don't have to read some of these ones
+    /*mHeaderItem.inkSize =*/ buffer.getInt();
+    /*mHeaderItem.linkOff =*/ buffer.getInt();
+    /*mHeaderItem.mapOff =*/ buffer.getInt();
+    mHeaderItem.setStringIdsSize(buffer.getInt());
+    mHeaderItem.setStringIdsOffset(buffer.getInt());
+    mHeaderItem.setTypeIdsSize(buffer.getInt());
+    mHeaderItem.setTypeIdsOffset(buffer.getInt());
+    mHeaderItem.setProtoIdsSize(buffer.getInt());
+    mHeaderItem.setProtoIdsOffset(buffer.getInt());
+    mHeaderItem.setFieldIdsSize(buffer.getInt());
+    mHeaderItem.setFieldIdsOffset(buffer.getInt());
+    mHeaderItem.setMethodIdsSize(buffer.getInt());
+    mHeaderItem.setMethodIdsOffset(buffer.getInt());
+    mHeaderItem.setClassDefsSize(buffer.getInt());
+    mHeaderItem.setClassDefsOffset(buffer.getInt());
+    /*mHeaderItem.dataSize =*/ buffer.getInt();
+    /*mHeaderItem.dataOff =*/ buffer.getInt();
```

Comment:
Can it be aligned so it is indented with same style

Reply:
The indentation can be fixed by using clang-format:
sudo apt install clang-format
clang-format --style=Google -i capstone/src/main/java/com/google/sps/data/dexparser/DexData.java

Reply:
It appeared well on my Cloudshell. I just think it's GitHub